spec/javascripts/modules/global-bar.spec.js

```diff
@@ -107,23 +114,30 @@ describe('Global bar module', function () {
     })
 
     it('tracks when dismiss link is clicked', function () {
-      globalBar.start(element)
+      globalBar = new GOVUK.Modules.GlobalBar(element[0])
+      globalBar.init()
 
-      $(element).find('.dismiss').click()
+      $(element).find('.dismiss')[0].click()
 
       expect(GOVUK.analytics.trackEvent).toHaveBeenCalledWith('Global bar', 'Manually dismissed', { nonInteraction: 1 })
     })
 
     it('tracks when clicking on a link marked with js-call-to-action', function () {
-      globalBar.start(element)
+      globalBar = new GOVUK.Modules.GlobalBar(element[0])
+      globalBar.init()
 
-      $(element).find('.js-call-to-action').click()
+      var link = $(element).find('.js-call-to-action')[0]
+      link.addEventListener('click', function (e) {
+        e.preventDefault()
+      })
```

Comment:
It wasn't clear to me why we need this event listener when we didn't before - perhaps that's something for the commit message?

Reply:
That's an excellent question. Honestly I don't know. I've just tested the original code to see the difference and I think it's something to do with the difference between calling `click` on a jQuery object compared with a JS DOM node. 

If you run this test in the browser (with the original code, not my changes) it leaves the element in question in the viewport. Calling `$('.js-call-to-action').click()` manually in the console triggers the relevant code but doesn't do anything else (expected). Calling `document.querySelector('.js-call-to-action').click()` (the vanilla JS equivalent) triggers the code but also navigates to the URL set in the href attribute for that link. So it seems that there's definitely a difference, but I'm not sure why this happens.

Reading the code my understanding is that this link should behave as a normal link - the only thing that the JS should do is fire a tracking event when it is clicked, so the logical thing for the test to do is to artificially prevent that navigation from happening. I don't know how the original code allowed that to be tested without causing a navigation to happen, but it seems the difference between the jQuery and native JS `click` events may account for it.

Reply:
Ah yes, I gave it a spin locally and the jQuery equivalent only triggers the event handlers and not the default action. I didn't realise that's how jQuery `.click` worked I thought it was the same and you needed to do `.trigger` for just event handlers.

Though looking [at the docs](https://api.jquery.com/click/#click-handler) they make it clear it's a shorthand for `.trigger('click')`